doc/designs/edns.md

```diff
@@ -0,0 +1,198 @@
+
+- Specify an Upstream DNS Server with DoT enabled: use the `--dot-forwarder` option to specify the upstream DNS server that supports DoT. The format must be 1.2.3.4#dns.server.test. You still need to specify at least one of `--forwarder`, `--auto-forwarders`, or `--no-forwarders` options for the non-encrypted communication as well as discovery process.
+
+- DoT Certificates. If you prefer to use certificates for DoT in Bind/Unbound, use the `--dns-over-tls-key` and `--dns-over-tls-cert` options. If these options are not specified, the IPA CA will be used to request a new certificate.
+
+- DNS Policy: use `--dns-policy` to define the DNS security policy for FreeIPA deployments. It accepts two values: `relaxed` and `enforced`. When set to `relaxed`, the system will attempt to use DoT but will fall back to unencrypted DNS if DoT is unavailable. This mode ensures compatibility with environments where encrypted DNS is not fully supported. When set to `enforced`, the system strictly requires DoT, and any DNS resolution that does not support encryption will be rejected, including discovery from clients.
+
+
+### CLI
+
+Overview of the CLI commands for the FreeIPA installers and FreeIPA DNS installers:
+
+Configuring DNS services using ipa-dns-install follows the same principles as installing DNS with the ipa-server-install utility.
+
+
+| Option | Description                                       |
+|:------------------------ | :------------------------------ |
+| --dns-over-tls  | enable DNS over TLS support. This option is present on both client and server. It deploys Unbound and configures BIND on the server to receive DoT requests.|
+| --dot-forwarder | the upstream DNS server with DoT support. It must be specified in the format 1.2.3.4#dns.server.test|
+| --dns-over-tls-key and --dns-over-tls-cert | in case user prefers to have the DoT certificate in BIND generated by themselves. If these are empty, IPA CA is used instead to request a new certificate. |
+| --dns-policy | Defines the DNS security policy. Accepts `relaxed` (attempts DoT but falls back to unencrypted DNS if unavailable) or `enforced` (strictly requires DoT and rejects unencrypted DNS resolution). |
+
+
+## Troubleshooting and debugging
+
+No DNS Resolution: 
+- Ensure that the DNS server (Bind or Unbound) is running.
+- Verify that the DNS server is listening on the correct IP addresses and ports.
+- Check the firewall settings to ensure that traffic on port 53 and port 853 is allowed.
+
+Encryption Not Working: 
+- Verify that forward-tls-upstream is enabled in Unbound.
+- Ensure that tls-cert-bundle or tls-system-cert is correctly configured and accessible.
+- Restart the Unbound service after making changes.
+
+Logs Not Showing Detailed Information:
+- Increase the verbosity level of logging in the configuration files for both BIND and Unbound.
+- Restart the respective services to apply the new logging levels.
+- Check the log files for detailed debug information.
```

Comment:
In general these troubleshooting steps require knowledge of where the logs are, where the configuration files are, what the expected IP addresses are supposed to be, how to find certificate bundles, etc.

Reply:
I refactored the troubleshooting section. Fixed.